### ISSUE TITLE
cabal should detect case errors in build-depends:

### ISSUE THREAD

**Issue by bos** [haskell/cabal#824]
(Imported from [Trac #834](http://hackage.haskell.org/trac/hackage/ticket/834), reported by cmh on 2011-04-22)

for example, with a foo.cabal file containing:

<blockquote>
build depends: http &gt;= 4000

</blockquote>

Resolving dependencies...
cabal: cannot configure foo-0.0.1. It requires http &gt;=4000
There is no available version of http that satisfies &gt;=4000

The solution is of course to specify HTTP, but the user may be led to believe that there is a version problem, which is confusing.

Perhaps a message such as - "Did you mean package HTTP" - Would be appropriate.

**Comment by BardurArantsson** [comment]
Seems very complicated for very little gain: Almost all packages of any note use exclusively lowercase these days AFAICT.

Close?

**Comment by phadej** [comment]
IMHO this shouldn't easy to someone to implement who wants to learn about solver

**Comment by BardurArantsson** [comment]
Are you going to do it?

Rhetorical question: My point is that it hasn't happened in the 4 years since this issue was raised in the original Trac(?). Therefore it isn't likely to happen ever based on the presence of this task in the tracker.

**Comment by phadej** [comment]
IMHO it's better to have issues open as an *idea pool*. You never know when new contributor traverses thru the list. It seems there are many new ones lately!

Btw. this issue reminds me about the behaviour that `cabal install quickcheck` do install `QuickCheck`. Yet we probably want exact case matches in cabal files, so adding new (ambigious) packages into hackage can't break old ones. This is probably why this issue felt odd.

**Comment by BardurArantsson** [comment]
No, it's not. Trust me on this. It's nothing but frustration and invisibility for *actually* important* issues. (I don't mean to say that "few issues => more contributors"... but what I can say is that "more issues => fewer contributors"... unless you're extremely popular like Mozilla/Firefox or Facebook/React.)

Btw, I was going to actually contribute some more stuff (hopefully I did at few bits & bobs, still pending PRs), but 500+ issues is not helping anyone.

In this particular case: Nobody's going to think "what a great idea, I must implement that!" when they see this issue. This issue is not in the realm of "great ideas". Revamping and improving the UI/UX completely as @tibbe seems to be going about is a great idea. This is a minor tweak.

I was not aware that "cabal install xxx" actually does case folding. That's incredible -- and somewhat broken in that it's actually *invalid* if you're running on a case sensitive file system. No changing it now, I suppose.

Still want this closed :)
